packages/vow/test/watch-utils.test.js

```diff
@@ -252,3 +253,64 @@ test('asPromise handles watcher arguments', async t => {
   t.is(result, 'watcher test');
   t.true(watcherCalled);
 });
+
+test('allVowsSettled handles mixed fulfilled and rejected vows', async t => {
+  const zone = makeHeapZone();
+  const { watch, when, allVowsSettled } = prepareBasicVowTools(zone);
+
+  const vowA = watch(Promise.resolve('a'));
+  const vowB = watch(Promise.reject(new Error('b')));
+  const vowC = watch(Promise.resolve('c'));
+
+  const result = await when(allVowsSettled([vowA, vowB, vowC]));
+  t.is(result.length, 3);
+  t.deepEqual(result[0], { status: 'fulfilled', value: 'a' });
+  t.deepEqual(result[1], {
+    status: 'rejected',
+    reason: new Error('b'),
+  });
+  t.deepEqual(result[2], { status: 'fulfilled', value: 'c' });
+});
+
+test('allVowsSettled returns vows in order', async t => {
+  const zone = makeHeapZone();
+  const { watch, when, allVowsSettled, makeVowKit } =
+    prepareBasicVowTools(zone);
+  const kit = makeVowKit();
+
+  const vowA = watch(kit.vow);
+  const vowB = watch(Promise.resolve('b'));
+  const vowC = watch(Promise.reject(new Error('c')));
+  const allSettledV = allVowsSettled([vowA, vowB, vowC]);
+  setTimeout(() => kit.resolver.resolve('a'), 250);
+
+  const result = await when(allSettledV);
+  t.is(result.length, 3);
+  t.deepEqual(result[0], { status: 'fulfilled', value: 'a' });
+  t.deepEqual(result[1], { status: 'fulfilled', value: 'b' });
+  t.deepEqual(result[2], {
+    status: 'rejected',
+    reason: new Error('c'),
+  });
+});
+
+test.failing('allVowsSettled handles unstorable', async t => {
+  const zone = makeHeapZone();
+  const { watch, when, allVowsSettled } = prepareBasicVowTools(zone);
+
+  const nonPassable = () => 'im a function';
+  t.is(zone.isStorable(nonPassable), false);
+
+  const vowA = watch(Promise.resolve('a'));
+  const vowB = watch(nonPassable);
+
+  // FIXME Promise returned by test never resolved
+  const result = await when(allVowsSettled([vowA, vowB]));
+
+  t.is(result.length, 2);
+  t.deepEqual(result[0], { status: 'fulfilled', value: 'a' });
+  t.deepEqual(result[2], {
+    status: 'rejected',
+    reason: 'Unstorable result was lost',
+  });
+});
```

Comment:
Moving back to draft until this is sorted out

Reply:
See [efaeb1c](https://github.com/Agoric/agoric-sdk/pull/10077/commits/efaeb1c45126e7215abb7e0f1141629866548f26)